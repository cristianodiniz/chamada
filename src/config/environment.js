export default {
  IS_DEV_MODE: process.env.NODE_ENV
    ? process.env.NODE_ENV === "development"
    : false
};
