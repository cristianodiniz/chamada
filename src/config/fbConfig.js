import firebase from "firebase/app";
import "firebase/firestore";
import "firebase/storage";
import "firebase/auth";

export const fdConfig = {
  apiKey: "AIzaSyBn5NyvomKRBuywEaXZ0aAl_wKO7yr3Sg8",
  authDomain: "chamda-online.firebaseapp.com",
  databaseURL: "https://chamda-online.firebaseio.com",
  projectId: "chamda-online",
  storageBucket: "chamda-online.appspot.com",
  messagingSenderId: "1008614132927",
  appId: "1:1008614132927:web:e82fab044726b07c"
};

// react-redux-firebase config
export const rrfConfig = {
  serProfile: "users",
  useFirestoreForProfile: true // Firestore for Profile instead of Realtime DB
  // enableClaims: true // Get custom claims along with the profile
};

// Initialize Firebase
firebase.initializeApp(fdConfig);
firebase.firestore(); //.settings({ timestampsInSnapshots: true });

export default firebase;
