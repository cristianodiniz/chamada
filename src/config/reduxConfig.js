import React from "react";
import { Provider } from "react-redux";

import { BrowserRouter as Router } from "react-router-dom";

import { ReactReduxFirebaseProvider } from "react-redux-firebase";
import { createFirestoreInstance } from "redux-firestore";

import firebase, { rrfConfig } from "./fbConfig";
import createReduxStore from "../store";

const store = createReduxStore();


const rrfProps = {
  firebase,
  config: rrfConfig,
  dispatch: store.dispatch,
  createFirestoreInstance
};

export function renderWithRedux(ui) {
  return (
    <Provider store={store}>
      <ReactReduxFirebaseProvider {...rrfProps}>
        <Router>{ui}</Router>
      </ReactReduxFirebaseProvider>
    </Provider>
  );
}
