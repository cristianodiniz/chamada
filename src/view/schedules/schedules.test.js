import React from "react";
import Schedules from "./schedules";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";

import { renderWithRedux } from "../../config/reduxConfig";

describe("Schedule Tests Component", function() {
  it("Should render without crashing", async done => {
    const div = document.createElement("div");
    act(() => {
      ReactDOM.render(renderWithRedux(<Schedules />), div);
    });
    ReactDOM.unmountComponentAtNode(div);
    done();
  }, 10000);

  test("Should Match Snapshot", async done => {
    let component
    act(() => {
      component = renderer.create(renderWithRedux(<Schedules />));
    });
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  }, 10000);
});
