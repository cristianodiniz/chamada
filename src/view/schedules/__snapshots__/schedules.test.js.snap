// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule Tests Component Should Match Snapshot 1`] = `
Array [
  <ul
    className="MuiList-root Schedules-root-40 MuiList-padding"
  />,
  <a
    className="Schedules-link-41"
    href="/schedule/add"
    onClick={[Function]}
  >
    <button
      aria-label="add"
      className="MuiButtonBase-root MuiFab-root Schedules-fab-42 MuiFab-primary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </a>,
]
`;
